src/NuGet.Services.Metadata.Catalog.Monitoring/Validation/Test/ValidationSourceRepository.cs

```diff
@@ -0,0 +1,34 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using NuGet.Configuration;
+using NuGet.Protocol;
+using NuGet.Protocol.Core.Types;
+
+namespace NuGet.Services.Metadata.Catalog.Monitoring
+{
+    /// <remarks>
+    /// This class exists to add some extra logic to the construction of <see cref="SourceRepository"/> using DI.
+    /// </remarks>
+    public class ValidationSourceRepository : SourceRepository
+    {
+        public ValidationSourceRepository(
+            PackageSource source, 
+            IEnumerable<INuGetResourceProvider> providers, 
+            FeedType type)
+            : base(
+                  source, 
+                  // We must add the default V3 providers to the providers we are injecting.
+                  Repository.Provider.GetCoreV3()
+                    .Concat(
```

Comment:
Could we instead register all of `GetCoreV3`'s resource providers into the dependency injection container?

Reply:
so register them keyed as both? i like that, it sounds cleaner